File: 2022/python/1/part2.py
groups = []

with open('input.txt') as file:
    current_group = []
    for line in file:
        if not line.strip():
            groups.append(current_group)
            current_group = []
        else:
            current_group.append(int(line))

summed_groups = [sum(group) for group in groups]
sorted_groups = sorted(summed_groups)
print(sum(sorted_groups[-3:]))

File: 2022/go/1/part2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func main() {
	file, _ := os.Open("input.txt")
	reader := bufio.NewScanner(file)

	totals := make([]int, 0)
	current := 0
	var num int
	more := false
	for {
		more = reader.Scan()
		line := reader.Text()

		if line != "" {
			num, _ = strconv.Atoi(line)
			current += num
		} else if current != 0{
			totals = append(totals, current)
			current = 0
		}

		if !more {
			break
		}
	}
	
	sort.Ints(totals)
	n := len(totals)
	fmt.Println(totals[n-1] + totals[n-2] + totals[n-3])
}